### PR TITLE
Shifting items from session to local storage and persisting the logged in user across tabs.

### PR DESCRIPTION
**Pull request by ashishbeehyv**

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/pages/citizen/SearchApp.js [hunk 1]
```diff
@@ -21,33 +21,33 @@ const Search = ({ path }) => {
...
+    };
+
+    setPayload(
+      Object.keys(data)
+        .filter((k) => data[k])
+        .reduce((acc, key) => ({ ...acc, [key]: typeof data[key] === "object" ? data[key] : data[key] }), {})
+    );
+  }
+  useEffect(() => {
+    const storedPayload = Digit.SessionStorage.get("AUDIT_APPLICATION_DETAIL") || {};
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/pages/citizen/SearchApp.js]: n/a

File: frontend/micro-ui/web/micro-ui-internals/example/src/index.js [hunk 2]
```diff
@@ -29,21 +29,21 @@
...
       access_token: token,
       info: userType !== "CITIZEN" ? JSON.parse(employeeInfo) : citizenInfo,
     });
   } else {
-    // if (!window.Digit.SessionStorage.get("User")?.extraRoleInfo) window.Digit.SessionStorage.set("User", { access_token: token, info: citizenInfo });
+    // if (!window.Digit.SessionStorage.get("User")?.extraRoleInfo) localStorage.setItem("User", { access_token: token, info: citizenInfo });
   }
 
-  window.Digit.SessionStorage.set("Citizen.tenantId", citizenTenantId);
+  localStorage.setItem("Citizen.tenantId", citizenTenantId);
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/example/src/index.js]: not needed

File: frontend/micro-ui/web/micro-ui-internals/example/src/index.js [hunk 3]
```diff
@@ -29,21 +29,21 @@
...
 
   if (userType !== "CITIZEN") {
-    window.Digit.SessionStorage.set("User", {
+    localStorage.setItem("User", {
       access_token: token,
       info: userType !== "CITIZEN" ? JSON.parse(employeeInfo) : citizenInfo,
     });
   } else {
-    // if (!window.Digit.SessionStorage.get("User")?.extraRoleInfo) window.Digit.SessionStorage.set("User", { access_token: token, info: citizenInfo });
+    // if (!window.Digit.SessionStorage.get("User")?.extraRoleInfo) localStorage.setItem("User", { access_token: token, info: citizenInfo });
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/example/src/index.js]: not needed

File: frontend/micro-ui/web/micro-ui-internals/example/src/index.js [hunk 4]
```diff
@@ -29,21 +29,21 @@
...
   }
 
-  window.Digit.SessionStorage.set("Citizen.tenantId", citizenTenantId);
+  localStorage.setItem("Citizen.tenantId", citizenTenantId);
 
-  if (employeeTenantId && employeeTenantId.length) window.Digit.SessionStorage.set("Employee.tenantId", employeeTenantId);
+  if (employeeTenantId && employeeTenantId.length) localStorage.setItem("Employee.tenantId", employeeTenantId);
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/example/src/index.js]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/Module.js [hunk 5]
```diff
@@ -151,6 +151,67 @@ const componentsToRegister = {
...
+    del: (key) => {
+      if (localStoreSupport()) {
+        storageClass.removeItem(k(key));
+      } else if (typeof window !== "undefined") {
+        window.eGov = window.eGov || {};
+        window.eGov.Storage = window.eGov.Storage || {};
+        delete window.eGov.Storage[k(key)];
+      }
+    },
+  });
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/Module.js:204]: not needed

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/registration/TermsCondition.js [hunk 6]
```diff
@@ -79,7 +79,7 @@ const TermsCondition = ({ t, config, params, setParams, pathOnRefresh }) => {
                   getUserDetails(refreshToken).then((res) => {
                     const { ResponseInfo, UserRequest: info, ...tokens } = res;
                     const user = { info, ...tokens };
-                    window?.Digit.SessionStorage.set("citizen.userRequestObject", user);
+                    localStorage.setItem("citizen.userRequestObject", user);
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/registration/TermsCondition.js:82]: n/a

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/registration/TermsCondition.js [hunk 7]
```diff
@@ -254,7 +254,7 @@
                       getUserDetails(refreshToken).then((res) => {
                         const { ResponseInfo, UserRequest: info, ...tokens } = res;
                         const user = { info, ...tokens };
-                        window?.Digit.SessionStorage.set("citizen.userRequestObject", user);
+                        localStorage.setItem("citizen.userRequestObject", user);
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/registration/TermsCondition.js:257]: n/a

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/registration/TermsCondition.js [hunk 8]
```diff
@@ -161,7 +161,7 @@
               getUserDetails(refreshToken).then((res) => {
                 const { ResponseInfo, UserRequest: info, ...tokens } = res;
                 const user = { info, ...tokens };
-                window?.Digit.SessionStorage.set("citizen.userRequestObject", user);
+                localStorage.setItem("citizen.userRequestObject", user);
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/citizen/registration/TermsCondition.js:164]: n/a

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/Module.js [hunk 9]
```diff
@@ -151,6 +151,67 @@
 };
 
 export const initCoreComponents = () => {
+  const localStoreSupport = () => {
+    try {
+      return "sessionStorage" in window && window["sessionStorage"] !== null;
+    } catch (e) {
+      return false;
+    }
+  };
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/Module.js:160]: done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/Module.js [hunk 10]
```diff
@@ -151,6 +151,67 @@
...
+      }
+    },
+  });
+  const Storage = getStorage(window.localStorage);
+
+  const setupLibraries = (Library, props) => {
+    window.Digit = window.Digit || {};
+    window.Digit[Library] = window.Digit[Library] || {};
+    window.Digit[Library] = { ...window.Digit[Library], ...props };
+  };
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/Module.js:212]: done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/Module.js [hunk 11]
```diff
@@ -151,6 +151,68 @@ const componentsToRegister = {
...
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/Module.js]: done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/Module.js [hunk 12]
```diff
@@ -151,6 +151,68 @@
...
+    del: (key) => {
+      if (localStoreSupport()) {
+        storageClass.removeItem(k(key));
+      } else if (typeof window !== "undefined") {
+        window.eGov = window.eGov || {};
+        window.eGov.Storage = window.eGov.Storage || {};
+        delete window.eGov.Storage[k(key)];
+      }
+    },
+  });
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/Module.js:205]: n/a

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/pages/citizen/SearchApp.js [hunk 13]
```diff
@@ -21,33 +21,33 @@
...
+      ..._data,
+      fromDate: convertDateToEpoch(_data?.fromDate),
+      toDate: convertDateToEpoch(_data?.toDate),
+    };
+
+    setPayload(
+      Object.keys(data)
+        .filter((k) => data[k])
+        .reduce((acc, key) => ({ ...acc, [key]: typeof data[key] === "object" ? data[key] : data[key] }), {})
+    );
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/core/src/pages/citizen/SearchApp.js]: done

**Review by anirudh-0 — APPROVED** [review]

**Pull request merged**